File: Server/runes/SetAttack.js
var entities = require('../entityManager');


exports.execute = function(rune, state) {
        var card = entities.getEntity(rune["cardGuid"], state);
        var amount = rune["amount"];
        card["totalAttack"] = amount;
}

exports.CreateRune = function(targetGuid, amount)
{
    var rune = {
        "runeType":"SetAttack",
        "cardGuid":targetGuid,
        "amount":amount
    }
    
    return rune;
}

exports.canSee = function (params) {
    return true;
}

File: Assets/Scripts/Runes/SetAttack.cs
using UnityEngine;
using System.Collections;

public class SetAttack : Rune {

	public string card_uid { get; set; }
	public int amount { get; set; }

	public override void Execute (System.Action action)
	{
		var ent = EntityManager.Singelton.GetEntity (card_uid) as damageable;
		ent.SetAttack (amount);
		action ();
	}

	public override void OnGUI ()
	{
		
	}
}
